src/Meziantou.AspNetCore.Components/GenericFormField.cs

```diff
@@ -75,6 +75,22 @@ public string DisplayName
             }
         }
 
+        public int DisplayOrder
+        {
+            get
+            {
+                var displayAttribute = Property.GetCustomAttribute<DisplayAttribute>();
+                if (displayAttribute != null)
+                {
+                    var displayOrder = displayAttribute.GetOrder();
+                    if (displayOrder != null)
+                        return (int)displayOrder;
+                }
+                //Undefined DisplayOrder are shown at the end 
+                return int.MaxValue;
```

Comment:
According to the documentation, we should use 10000 when no value is set

https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.displayattribute.getorder?view=net-5.0#remarks

```suggestion

                // https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.displayattribute.getorder?view=net-5.0#remarks
                // If an order is not specified, presentation layers should consider setting the value 
                // of the Order property to 10000. This value lets explicitly-ordered fields be displayed 
                // before and after the fields that do not have a specified order.
                return 10_000;
```